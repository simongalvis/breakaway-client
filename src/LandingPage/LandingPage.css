.LandingPage{
    margin-bottom: 80px;
    display: flex;
    flex-direction: column;
    min-height: 520px;
}
.hero{
    font-size: 33px;
    margin: 24px 0px; 
    padding-left:16px;
    padding-right:16px;
}
#landingpage-ui-image{
    height:625px;
    width:342px;
    align-self: center;
}
#landingpage-signup-button{
    margin: 24px 24px;
    width: 290px;
    border-radius: 33px;
    background-color: rgb(13, 146, 255);
    color: white;
    border: 1px solid white;
}
#hero-text{
    font-size: 27px;
    color: gray;
    margin: 24px 0px; 
    padding-left:16px;
    padding-right:16px;
    max-width: 700px;
    
    align-self: center;
    /* text-shadow: 0px 0px 1px black; */

}