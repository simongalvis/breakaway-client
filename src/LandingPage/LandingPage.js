import React from 'react';
import {Route, Link } from 'react-router-dom';
import './LandingPage.css';
import NavBar from '../NavBar/NavBar';
import sampleHero from '../images/sample-hero.jpg'

class LandingPage extends React.Component{


    render(){
        return(
            
            <div className='LandingPage'>
                
                <h2 className='hero'>The #1 way to track your athletic progress</h2>
                <img src={sampleHero} id="landingpage-ui-image" alt="Breakaway ui on mobile"/>
                <Link to='/signup-page'>
                    <button id='landingpage-signup-button'>Sign up</button>
                </Link>
                

            </div>
            
        )
    }
}

export default LandingPage;