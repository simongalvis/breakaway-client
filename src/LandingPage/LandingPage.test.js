import React from 'react'
import ReactDOM from 'react-dom'
import LandingPage from './LandingPage'
import renderer from 'react-test-renderer';

describe(`LandingPage component`, () =>{
  it(`renders without crashing`, () =>{
    const div = document.createElement('div');
  
    ReactDOM.render(<LandingPage/>, div);
  
    ReactDOM.unmountComponentAtNode(div);
  })  
  it(`Renders UI as expected`, () =>{
    const tree = renderer
      .create(<LandingPage/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })
})
