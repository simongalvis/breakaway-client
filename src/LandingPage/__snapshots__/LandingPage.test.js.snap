// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage component Renders UI as expected 1`] = `
<div
  className="LandingPage"
>
  <h2
    className="hero"
  >
    The simple way to track your athletic progress
  </h2>
  <p
    id="hero-text"
  >
    Breakaway allows you to log your runs, walks, hikes, and other activities in one convenient location
  </p>
  <img
    alt="Breakaway ui on mobile"
    id="landingpage-ui-image"
    src="sample-hero.jpg"
  />
  <a
    href="/signup-page"
    onClick={[Function]}
  >
    <button
      id="landingpage-signup-button"
    >
      <b>
        Sign Up
      </b>
    </button>
  </a>
</div>
`;
